packages/docs-website/docs/docs/protocol-tutorial/off-chain-funding.md

```diff
@@ -320,6 +320,338 @@ classDef defunded opacity:0.2;
+  participants: [Alice, Bob, Hub]
+};
+const GA: Channel = {
+  chainId,
+  channelNonce: BigNumber.from(0).toHexString(),
+  participants: [Alice, Hub]
+};
+const GB: Channel = {
+  chainId,
+  channelNonce: BigNumber.from(0).toHexString(),
+  participants: [Bob, Hub]
+};
+const commonFields = {
+  isFinal: false,
+  appDefinition: AddressZero,
+  appData: HashZero,
+  challengeDuration: 86400, // 1 day
+  };
+
+
+// Joint channel
+const JPreFS: State = {
+  ...commonFields
+  channel: X,
+  outcome: [
+    {
+      asset: MAGIC_ADDRESS_INDICATING_ETH,
+      allocationItems: [
+        {destination: aliceDestination, amount: parseUnits(a, 'wei').toHexString()},
+        {destination: bobDestination, amount: parseUnits(b, 'wei').toHexString()}
+        {destination: hubDestination, amount: parseUnits(a+b, 'wei').toHexString()}
+      ]
+    }
+  ],
+  turnNum: 1
+};
+
+// Collect a support proof by getting all participants to sign this state
+signState(JPreFS, aliceSigningKey);
+signState(JPreFS, bobSigningKey);
+signState(JPreeFS, hubSigningKey);
+
+// Guarantor channels
+const GAPreFS: State = {
+  ...commonFields
+  channel: GA,
+  outcome: [
+    {
+      asset: MAGIC_ADDRESS_INDICATING_ETH,
+      guarantee: {
+        targetChannelId: getChannelId(J)
+        destinations: [Alice, XChannelId, Hub]
+      }
+    }
+  ],
+  turnNum: 1
+};
+const GBPreFS: State = {
+  ...commonFields
+  channel: GB,
+  outcome: [
+    {
+      asset: MAGIC_ADDRESS_INDICATING_ETH,
+      guarantee: {
+        targetChannelId: getChannelId(J)
+        destinations: [Bob, XChannelId, Hub]
+      }
+    }
+  ],
+  turnNum: 1
+};
+
+// Collect a support proof by getting all participants to sign this state
+signState(GAPreFS, aliceSigningKey);
+signState(GAPreFS, hubSigningKey);
+signState(GBPreFS, aliceSigningKey);
+signState(GBPreFS, hubSigningKey);
+
+```
+
+The guarantor channels have a special outcome called a [guarantee](./outcomes#outcomes-that-guarantee), which we show as a dashed arrow:
+
+<Mermaid chart='
+graph LR;
+linkStyle default interpolate basis;
+ETHAssetHolder( )
+appChannel((X)):::defunded
+J((J)):::defunded
+GA((GA)):::defunded
+GB((GB)):::defunded
+ledgerA((LA))
+Alice(( )):::alice
+hub(( )):::hub
+ETHAssetHolder-->|a+b|ledgerA;
+ledgerA-->|a|Alice;
+ledgerA-->|b|hub;
+ledgerB((LB))
+Bob(( )):::bob
+ETHAssetHolder-->|a+b|ledgerB;
+ledgerB-->|a|hub;
+ledgerB-->|b|Bob;
+appChannel-->|a|Alice
+appChannel-->|b|Bob
+J-->|a+b|hub
+J-->|a|Alice
+J-->|b|Bob
+GA-.->|AXH|J
+GB-.->|BXH|J
+classDef hub fill:#f96
+classDef alice fill:#eb4034
+classDef bob fill:#4e2bed
+linkStyle 6,7,8,9,10,11,12 opacity:0.2
+linkStyle 11,12 stroke-dasharray: 5 5,opacity:0.2
+classDef defunded opacity:0.2;
+' />
+
+Now, we "plug in" `GA` by directing the flow of funds away from the "end users" and into the network of channels. We do the same for `GB`: and in fact can do this in any order. We should think of `J` being funded _only_ when both guarantees have been updated.
```

Comment:
So netlify will build a preview for the docs website, but the URL is not that easy to find. You have to go into circle CI, drill down into the `deploy-docs-website` step, and inspect the logs at the end.

For example https://611e6707bf6734082ac1df3a--nitro-protocol.netlify.app/docs/protocol-tutorial/off-chain-funding#virtual-funding 

(I'm not yet happy with these diagrams, because they are too busy and a bit of a mess. I think we can improve them, though.)

As for generating a local preview, once the monorepo is installed and linked etc, a `yarn start` from the `docs-website` directory should work (it is not expected to fail).

Reply:
About docs.statechannels.org: the automatic builds via the netlify GUI are indeed disabled. However, if a PR is merged, circle will build and redeploy the site via the netlify CLI (hence it is up to date with `master`).

Reply:
You can also preview mermaid diagrams by pasting the markup into https://mermaid-js.github.io/mermaid-live-editor/